### PR TITLE
Multischema endpoint

### PR DESCRIPTION
**Pull request by ceesem**
Adding the multischema download endpoints and using them in the tables/views interface. Wound up refactoring the materialization initialization to use multithreading for pinging the server for various information, but on the plus side the .tables and .views interfaces are populated at the same time with minor overhead cost. Also, added small tweaks that seem to improve tab-autocompletion for these services as well.

I think this is working, but there are some changes to client.materialize that could be significant if they don't work as expected, so a bit of caution is warranted.

### CODE REVIEW

**Comment by fcollman** [comment]
If some caution is needed perhaps it’s time we added some tests for this

**Comment by ceesem** [comment]
Agreed, will add tests before this gets merged. Although what I was concerned about should be caught by the materialization client tests, if I understand them correctly.

**Comment by ceesem** [comment]
Tests added for the table manager. Not perfect because they have to mock the metadata one gets from the server, but at least there's an example there to compare against.

File: tests/test_tablemanager.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    tables = TableManager(None, metadata=table_meta, schema=schema_meta)
+    qry = tables.allen_column_mtypes_v2(pt_root_id=[123, 456], target_id=271700)
+    params = qry.filter_kwargs_live
+    assert len(tables)==2
+    assert 'allen_column_mtypes_v2' in params.get('filter_equal_dict')
+    assert 'nucleus_detection_v0' in params.get('filter_in_dict')
+    assert 'allen_column_mtypes_v2' == qry.joins_kwargs.get('joins')[0][0]
+
+def test_view_manager():
+    views = ViewManager(None, view_metadata=views_list, view_schema=views_schema)
```
Comment (fcollman) [inline tests/test_tablemanager.py:174]: this should be done with responses so it tests that portion of the code

Comment (ceesem) [reply tests/test_tablemanager.py:174]:
```        with ThreadPoolExecutor(max_workers=4) as executor:
            metadata.append(
                executor.submit(
                    self.get_tables_metadata,
                )
            )
            metadata.append(
                executor.submit(
                    self.fc.schema.schema_definition_all
                )
            )
            metadata.append(
                executor.submit(
                    self.get_views
                )
            )
            metadata.append(
                executor.submit(
                    self.get_view_schemas
                )
            )
        if self.fc is not None:
            tables = TableManager(self.fc, metadata[0].result(), metadata[1].result())
        else:
            tables = None
        self.tables = tables
```

This is how the code is supplied to table manager now. I could do this with responses, but won't I just be plugging the answer directly in anyway?

Comment (ceesem) [reply tests/test_tablemanager.py:174]: Fair, changed now.

**Review by fcollman — APPROVED** [review]

**Pull request merged**